hack/cluster-sync-handler.sh

```diff
@@ -8,21 +8,38 @@ ${KUBECTL} delete --ignore-not-found -f ${local_handler_manifest}
 # Set debug verbosity level for logs when using cluster-sync
 sed "s#--v=production#--v=debug#" ${local_handler_manifest} | ${KUBECTL} create -f -
 
-for i in {300..0}; do
-    # We have to re-check desired number, sometimes takes some time to be filled in
-    desiredNumberScheduled=$(${KUBECTL} get daemonset -n nmstate nmstate-handler -o=jsonpath='{.status.desiredNumberScheduled}')
+function getDesiredNumberScheduled {
+        echo $(${KUBECTL} get daemonset -n nmstate $1 -o=jsonpath='{.status.desiredNumberScheduled}')
+}
+
+function getNumberAvailable {
+        echo $(${KUBECTL} get daemonset -n nmstate $1 -o=jsonpath='{.status.numberAvailable}')
+}
+
+function isOk {
+        desiredNumberScheduled=$(getDesiredNumberScheduled $1)
+        numberAvailable=$(getNumberAvailable $1)
 
-    numberAvailable=$(${KUBECTL} get daemonset -n nmstate nmstate-handler -o=jsonpath='{.status.numberAvailable}')
 
-    if [ "$desiredNumberScheduled" == "$numberAvailable" ]; then
-        echo "nmstate-handler DS is ready"
-        break
+        if [ "$desiredNumberScheduled" == "$numberAvailable" ]; then
+          echo "$1 DS is ready"
+          return 0
+        else
+          return 1
+        fi
```

Comment:
not needed the following is enough, since the lresult code of a function is from the last command in a function so just comparing stuff is enough.

```bash
[ "$desiredNumberScheduled" == "$numberAvailable" ]
```

Reply:
Hi @qinqon, maybe I don't understand your comment. But the function is not working without the return values. The result code of [ "$desiredNumberScheduled" == "$numberAvailable" ] is 0 whether the condition value is true or false.